src/lib/seam/thermostats/use-update-thermostat.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  useMutation,
+  type UseMutationResult,
+  useQueryClient,
+} from '@tanstack/react-query'
+import type {
+  SeamError,
+  ThermostatDevice,
+  ThermostatsListResponse,
+  ThermostatUpdateRequest,
+} from 'seamapi'
+
+import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
+
+// UPSTREAM: Missing ThermostatUpdateResponse in seamapi.
+type UseUpdateThermostatData = Record<string, unknown>
+type UseUpdateThermostatMutationParams = ThermostatUpdateRequest
+
+export function useUpdateThermostat(): UseMutationResult<
+  UseUpdateThermostatData,
+  SeamError,
+  UseUpdateThermostatMutationParams
+> {
+  const { client } = useSeamClient()
+  const queryClient = useQueryClient()
+
+  return useMutation<
+    UseUpdateThermostatData,
+    SeamError,
+    UseUpdateThermostatMutationParams
+  >({
+    mutationFn: async (mutationParams: UseUpdateThermostatMutationParams) => {
+      if (client === null) throw new NullSeamClientError()
+
+      return await client.thermostats.update(mutationParams)
+    },
+    onSuccess: (_data, variables) => {
+      queryClient.setQueryData<ThermostatDevice | null>(
+        ['thermostats', 'get', { device_id: variables.device_id }],
+        (thermostat) => {
+          if (thermostat == null) {
+            return
+          }
+
+          return getUpdatedThermostat(thermostat, variables)
+        }
+      )
+
+      queryClient.setQueryData<ThermostatsListResponse['thermostats']>(
+        ['thermostats', 'list', { device_id: variables.device_id }],
+        (thermostats): ThermostatDevice[] => {
+          if (thermostats == null) {
+            return []
+          }
+
+          return thermostats.map((thermostat) => {
+            if (thermostat.device_id === variables.device_id) {
+              return getUpdatedThermostat(thermostat, variables)
+            }
+
+            return thermostat
+          })
+        }
+      )
+    },
+  })
+}
+
+function getUpdatedThermostat(
+  thermostat: ThermostatDevice,
+  variables: UseUpdateThermostatMutationParams
+): ThermostatDevice {
+  return {
+    ...thermostat,
+
+    properties: {
+      ...thermostat.properties,
+
+      default_climate_setting: {
+        ...thermostat.properties.default_climate_setting,
+        manual_override_allowed:
```

Comment:
Can you explain this logic a bit? Why do we default to false here vs. allow the null? Should we default to `variables.<something>.manual_override_allowed`?

Reply:
Yeah, I struggled with this for a bit. This is just to make TS happy. The problem is that the type for `manual_override_allowed ` when updating is `boolean | undefined`, but it has to be `boolean`—It won't allow `null`. We can also get around this by writing:

```
 default_climate_setting: {
  ...thermostat.properties.default_climate_setting,
  ...variables.default_climate_setting,
} as ClimateSetting
```

We could default to the definition in `variables`, but that would require an additional null-check and a subsequent default.

Reply:
I'm going to merge this in to unblock @Andrewjeska—logic can be adjusted if needed during implementation.

Reply:
> This is just to make TS happy

I trust the types! If we type cast we may actually introduce an unexpected null.